### PR TITLE
dApp: PostHog integration

### PR DESCRIPTION
**Pull request by kpyszkowski**
Closes: #747
Closes: AENG-30

### Overview
This PR integrates dApp with PostHog service to track users activities and analyze it. The PostHog project was created. All dApp activity is now accessible through [the dashboard](https://us.posthog.com/project/104474).
The API key and endpoint are available in the project settings.

### Features
- dApp integration using [PostHog's official React SDK](https://www.npmjs.com/package/posthog-js)
- Autocapture: all user interactions are tracked, any button click, any form fill
- Manually triggered `$pageview` capture event: autocapture for page view event was disabled to prevent false positive captures. It is triggered manually as user visits the dApp.
- User identification via wallet address
- Added custom events with additional payload information: 
  - `deposit_success` - payload: `{ transactionHash }`
  - `deposit_failure` - payload: `{ cause }`
  - `withdrawal_success` - wasn't able to easily get any reasonable payload here, can be done later
  - `withdrawal_failure` - payload: `{ cause }`
  - `points_claim_success` - payload: `{ claimedAmount, totalAmount }`
  - `points_claim_failure` - payload: `{ cause }`
- Defined feature flag to toggle PostHog integration

### Changes
- Added new env variables to define API key and endpoint and a feature flag
- Defined preconfigured `PostHogProvider` to inject the service into application
- Defined hooks: `usePostHogCapture`, `usePostHogIdentity`, `usePostHogPageViewCapture`
- Integrated key actions to capture PostHog custom events (see above)

### Caveats
To ensure the `$pageview` event is captured properly it's needed to call it as a side effect as the `location` value changes. To access this value (via `useLocation` hook) the router's context needs to be in scope of PostHog's context.
The only way to achieve it was to do it in the `Layout` component where both contexts are available.

### Note
It's impossible to exclude captures for development environment. To keep data logs clean and ensure it contains only user's data I suggest to **keep the feature flag disabled** while using development environment.

### CODE REVIEW

File: dapp/src/components/TransactionModal/ActiveStakingStep/DepositBTCModal.tsx [hunk 1]
```diff
@@ -63,8 +69,16 @@ export default function DepositBTCModal() {
...
         onError(error)
       }
+
+      const captureParameters =
+        error instanceof Error
+          ? {
+              cause: error.message,
+            }
+          : undefined
+      handleCapture(PostHogEvent.DepositFailure, captureParameters)
```
Comment (r-czajkowski) [inline dapp/src/components/TransactionModal/ActiveStakingStep/DepositBTCModal.tsx:79]: Maybe we should extract this code to a separate hook or util function? We do the same in `SignMessageModal.tsx` file.

Comment (kpyszkowski) [reply dapp/src/components/TransactionModal/ActiveStakingStep/DepositBTCModal.tsx]: Nice one
Ref commit: fa12d8a

**Comment by kpyszkowski** [comment]
@r-czajkowski 

> `withdrawal_success` - wasn't able to easily get any reasonable payload here, can be done later

Do you have any ideas about payload we could attach to the capture there? I mean something useful and available in the scope so can be used.

**Comment by r-czajkowski** [comment]
> @r-czajkowski
> 
> > `withdrawal_success` - wasn't able to easily get any reasonable payload here, can be done later
> 
> Do you have any ideas about payload we could attach to the capture there? I mean something useful and available in the scope so can be used.

Maybe we can add `transactionHash` and `redemptionKey` see https://github.com/thesis/acre/blob/main/sdk/src/modules/account.ts#L200.

**Comment by kpyszkowski** [comment]
@r-czajkowski 
> Maybe we can add `transactionHash` and `redemptionKey` 

It's not that straight forward. I need to play around with mutation function to pass these parameters to callbacks.
Such change might affect whole withdrawal flow. I don't want to risk it. It's doable but let's not block the PR.

I prepared POC which looks fine for further investigation:
[withdrawal-success-payload-poc.patch](https://github.com/user-attachments/files/17950424/withdrawal-success-payload-poc.patch)

**Review by r-czajkowski — APPROVED** [review]
Let's remember to set up correct env variables on testnet and production. I belive we should disable the posthog support on testnet.

**Comment by kpyszkowski** [comment]
Updated environmental variables:

<details>
  <summary>Mainnet (bitcoin.acre.fi)</summary>
  image
</details>

<details>
  <summary>Testnet (bitcoin.test.acre.fi)</summary>
  image
</details>

**Review by r-czajkowski — APPROVED** [review]

**Pull request merged**